FreeRTOS_Sockets.c

```diff
@@ -1297,6 +1297,9 @@ BaseType_t vSocketBind( FreeRTOS_Socket_t * pxSocket,
         struct freertos_sockaddr xAddress;
     #endif /* ipconfigALLOW_SOCKET_SEND_WITHOUT_BIND */
 
+    configASSERT( pxSocket != NULL );
+    configASSERT( pxSocket != FREERTOS_INVALID_SOCKET );
```

Comment:
Good catch to move these asserts to a location where `pxSocket` has not been used yet.